mole/tests/integration_tests.rs

```diff
@@ -0,0 +1,148 @@
+use honey_badger::{Auth, AuthLevel};
+use mole::ChannelStatePersistenceClient;
+use rand::Rng;
+use serial_test::{parallel, serial};
+use simplelog::TestLogger;
+use std::env;
+use std::sync::{Arc, Once};
+
+static INIT_LOGGER_ONCE: Once = Once::new();
+
+#[cfg(test)]
+#[ctor::ctor]
+fn init() {
+    INIT_LOGGER_ONCE.call_once(|| {
+        TestLogger::init(simplelog::LevelFilter::Info, simplelog::Config::default()).unwrap();
+    });
+}
+
+#[test]
+#[parallel]
+fn test_health_positive_check() {
+    let storage_client = build_storage_client();
+
+    assert!(storage_client.check_health());
+}
+
+#[test]
+#[serial]
+fn test_health_negative_check() {
+    let original_url = env::var("GRAPHQL_HEALTH_URL").unwrap();
+    std::env::set_var("GRAPHQL_HEALTH_URL", "https://localhost"); // No GraphQL endpoint here ...
+
+    let storage_client = build_storage_client();
+    assert!(!storage_client.check_health());
+
+    std::env::set_var("GRAPHQL_HEALTH_URL", original_url);
+}
+
+#[test]
+fn test_channel_monitor_field_exists() {
+    let client = build_storage_client();
+
+    assert!(client.verify_channel_monitor_field_exists().unwrap());
+}
+
+#[test]
+fn test_channel_manager_field_exists() {
+    let client = build_storage_client();
+
+    assert!(client.verify_channel_manager_field_exists().unwrap());
+}
+
+#[test]
+fn test_channel_monitor_persistence() {
+    let client = build_storage_client();
+
+    // Channel 1: Dummy data
+    let channel_id_0 = "11111111111111111111111111111111111111111111111111111111111111110000";
+    let channel_monitors_0 = vec![
+        rand::thread_rng().gen::<[u8; 32]>().to_vec(),
+        rand::thread_rng().gen::<[u8; 32]>().to_vec(),
+        rand::thread_rng().gen::<[u8; 32]>().to_vec(),
+    ]; // 3 encrypted channel monitors
+
+    // Channel 2: Dummy data
+    let channel_id_1 = "22222222222222222222222222222222222222222222222222222222222222228888";
+    let channel_monitor_1 = rand::thread_rng().gen::<[u8; 32]>().to_vec(); // Encrypted channel monitor
+
+    let inst_id = "integration_test"; // installation id = Usually the current version of 3L
+    let device = rand::thread_rng().gen::<[u8; 32]>().to_vec(); // Encrypted device info
+
+    // For channel 1, simulate 3 channel updates
+    for monitor in &channel_monitors_0 {
+        client
+            .write_channel_monitor(channel_id_0, &monitor, inst_id, &device)
+            .unwrap();
+    }
+
+    // For channel 2, only write 1 channel state
+    client
+        .write_channel_monitor(channel_id_1, &channel_monitor_1, inst_id, &device)
+        .unwrap();
+
+    let retrieved_channel_ids = client.get_channel_monitor_ids().unwrap();
+    assert_eq!(retrieved_channel_ids.len(), 2);
+    assert_ne!(retrieved_channel_ids[0], retrieved_channel_ids[1]);
+    for channel_id in retrieved_channel_ids {
+        if channel_id != channel_id_0 && channel_id != channel_id_1 {
+            panic!("Unexpected channel id: {}", channel_id);
+        }
+    }
```

Comment:
I think this would be a cleaner way of doing it 
```suggestion
    assert!(retrieved_channel_ids.contains(&String::from(channel_id_0)));
    assert!(retrieved_channel_ids.contains(&String::from(channel_id_1)));
```

Reply:
Check out a84b78f669db126e10e6ee340034b2edf70a7a3e, this also made the above assert-not-equal statement redundant.